src/main/scala/nl/knaw/dans/easy/fsrdb/FsRdbUpdater.scala

```diff
@@ -33,32 +36,54 @@ object FsRdbUpdater {
   val NS_EASY_FOLDER = "easy-folder"
   val namespaces = List(NS_EASY_FILE, NS_EASY_FOLDER)
 
-  def run(implicit s: Settings): Try[Unit] =
+  def run(implicit s: Settings): Try[Unit] = {
+    log.info(s"$s")
+    if (s.input.isDefined)
+      updateDatasets(Source.fromFile(s.input.get).getLines.toList)
+    else if (s.datasetPids.isDefined)
+      updateDatasets(s.datasetPids.get)
+    else
+      throw new IllegalArgumentException("No datasets specified to update")
+  }
+
+  private def updateDatasets(datasetPids: List[String])(implicit s: Settings): Try[Unit] = Try {
     for {
-      _ <- Try { FedoraRequest.setDefaultClient(new FedoraClient(s.fedoraCredentials)) }
-      _ = log.info(s"Checking if dataset ${s.datasetPid} exists")
-      _ <- existsDataset()
+      _ <- Try {FedoraRequest.setDefaultClient(new FedoraClient(s.fedoraCredentials))}
+      conn = DriverManager.getConnection(s.postgresURL)
```

Comment:
getConnection can throw an exception. turn `=` into `<- Try{`, of coursa add a `}` too.
This might apply to more lines.